### ISSUE TITLE
RF base model

### ISSUE THREAD

**Issue by Yazdan-Lotfi-Pour-1996** [Yazdan-Lotfi-Pour-1996/CAP-MLP-model#1]
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split, KFold, cross_val_score
from sklearn.ensemble import RandomForestRegressor
from sklearn.multioutput import MultiOutputRegressor
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.metrics import r2_score, mean_squared_error

df = pd.read_csv('/content/Final_Cold_Plasma_Data_300.csv')

X_features = ['X (mm)', 'Z (mm)', 'Voltage (kV)', 'Gas_Flow (L/min)', 'Frequency (kHz)', 'Gas_Composition', 'E_field (V/cm)']
y_features = ['ne (cm^-3)', 'Te (eV)', 'Ion_Pos_Density (cm^-3)', 'Ion_Neg_Density (cm^-3)',
              'O_density (cm^-3)', 'H_density (cm^-3)', 'H2O2_density (cm^-3)', 'Ar_metastable_density (cm^-3)']
X = df[X_features]
y = df[y_features]
numeric_features = ['X (mm)', 'Z (mm)', 'Voltage (kV)', 'Gas_Flow (L/min)', 'Frequency (kHz)', 'E_field (V/cm)']
categorical_features = ['Gas_Composition']

preprocessor = ColumnTransformer(
    transformers=[
        ('num', StandardScaler(), numeric_features),
        ('cat', OneHotEncoder(handle_unknown='ignore'), categorical_features)
    ])

base_model = RandomForestRegressor(n_estimators=100, random_state=42)
model = Pipeline(steps=[
    ('preprocessor', preprocessor),
    ('regressor', MultiOutputRegressor(base_model))
])

kf = KFold(n_splits=5, shuffle=True, random_state=42)
scores = cross_val_score(model, X, y, cv=kf, scoring='r2')
print("R² Scores across folds:", scores)
print("Average R²:", np.mean(scores))

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
model.fit(X_train, y_train)
y_pred = model.predict(X_test)

print("\n Evaluation on test set:")
for i, target in enumerate(y.columns):
    r2 = r2_score(y_test[target], y_pred[:, i])
    mse = mean_squared_error(y_test[target], y_pred[:, i])
    rmse = mse ** 0.5
    print(f"{target}: R² = {r2:.4f} | RMSE = {rmse:.4f}")
